lib/resty/rediscluster.lua

```diff
@@ -104,18 +174,21 @@ local function try_hosts_slots(self, serv_list)
     if #serv_list < 1 then
         return nil, "failed to fetch slots, serv_list config is empty"
     end
-
     for i = 1, #serv_list do
         local ip = serv_list[i].ip
         local port = serv_list[i].port
+        local is_healthy = is_node_healthy(ip, port)
+        if not is_healthy then
+            goto continue
+        end
         local redis_client = redis:new()
         local ok, err, max_connection_timeout_err
-        redis_client:set_timeouts(config.connect_timeout or DEFAULT_CONNECTION_TIMEOUT,
-                                  config.send_timeout or DEFAULT_SEND_TIMEOUT,
-                                  config.read_timeout or DEFAULT_READ_TIMEOUT)
-
         --attempt to connect DEFAULT_MAX_CONNECTION_ATTEMPTS times to redis
         for k = 1, config.max_connection_attempts or DEFAULT_MAX_CONNECTION_ATTEMPTS do
+            local attempt_timeout = k == 1 and 100 or 800  -- 100ms first attempt, 800ms retry
```

Comment:
Considering that the user's redis may have mTLS enabled, 100ms seems relatively small. 200ms for the first time and 700ms for the second time may be better.

Reply:
@nic-6443 What do you think?

Reply:
I think this parameter(connect timeout) only affects the time required for TCP's three-way handshake and has nothing to do with mTLS?

Reply:
My mistake. This should be just the TCP connection timeout, and has nothing to do with the MTLS handshake verification.

Reply:
after confirmed with @membphis  and @nic-chen , we decide to keep current hard-code timeout way, If our users need to customize this behavior in the future, we will provide configuration options. cc @Revolyssup